src/NPCManager.cpp

```diff
@@ -91,6 +91,119 @@ void NPCManager::init() {
     REGISTER_SHARD_PACKET(P_CL2FE_REQ_PC_WARP_USE_NPC, npcWarpHandler);
     REGISTER_SHARD_PACKET(P_CL2FE_REQ_NPC_SUMMON, npcSummonHandler);
     REGISTER_SHARD_PACKET(P_CL2FE_REQ_BARKER, npcBarkHandler);
+    REGISTER_SHARD_PACKET(P_CL2FE_REQ_PC_VENDOR_START, npcVendorStart);
+    REGISTER_SHARD_PACKET(P_CL2FE_REQ_PC_VENDOR_TABLE_UPDATE, npcVendorTable);
+    REGISTER_SHARD_PACKET(P_CL2FE_REQ_PC_VENDOR_ITEM_BUY, npcVendorBuy);
+}
+
+void NPCManager::npcVendorBuy(CNSocket* sock, CNPacketData* data) {
+    if (data->size != sizeof(sP_CL2FE_REQ_PC_VENDOR_ITEM_BUY))
+        return; // malformed packet
+
+    sP_CL2FE_REQ_PC_VENDOR_ITEM_BUY* req = (sP_CL2FE_REQ_PC_VENDOR_ITEM_BUY*)data->buf;
+    PlayerView& plr = PlayerManager::players[sock];
+
+    int itemCost = 100; //TODO: placeholder, look up the price of item
+
+    //TODO: need to check if user has enough money + free inventory space, else send fail packet 
```

Comment:
The fail packet is being handled below. Stale comment?

I agree that inventory manipulation should be independently validated, though. Check if `resp.iInvenSlotNum` overflows/underflows `plr->Inven`.